kuma/users/jinja2/socialaccount/includes/change_username.html

```diff
@@ -1,4 +1,5 @@
 <div id="change-username-container" {% if is_hidden %}class="hidden"{% endif %}>
     <label for="{{ form.username.id_for_label }}" class="visually-hidden">{{ form.username.label }}</label>
     {{ form.username }}
+    <div id="{{ form.username.id_for_label }}-error" class="error hidden">{{ _('A username is required.') }}</div>
```

Comment:
I still don't get it with these. If I try to submit the form with an empty username this happens:
<img width="559" alt="Screen Shot 2020-02-10 at 11 43 39 AM" src="https://user-images.githubusercontent.com/26739/74170318-b7749700-4bfa-11ea-93bf-2876f1746e07.png">
So how do you possibly benefit from this line here?

Reply:
How do you even get into that state 🤣 Did you add `novalidate` to the form? Also, how is the descriptive copy below the input field red? Hmm actually, it seems like the label, (server validation)error message, input field, and the descriptive copy is all inside a container with an `error` class.

Could you add the steps to reproduce? Thanks, Peter

Reply:
The native web form `required` on the `<input>` is easy to bypass. Just type an empty string (or even many spaces to empty string) into the form and submit. That'll send it into Django and thankfully, somewhere in our Django form handling, we do `username = username.strip()` before the validation kicks in.

Reply:
Gotcha!

Reply:
Ok, this is resolved as well. I removed the `error` class that was adding to the containing list element so, the styling will not be correct when a server error is returned.

With regards to the JS and the error message you are referring to here, I added a `trim` to the string that I should have done from the start so, now it should let you submit an empty username field.

As for the reason this was all added. There is/was? a scenario where you can land on the signup page with an empty username. In those instances, clicking "create profile" just seemed to do nothing as the actual input field was hidden.

This ensures that the field is made visible and an error message is presented to the user.